### PR TITLE
Export account

### PR DESCRIPTION
**Pull request by Anze1m**
This PR implements functionalities mentioned in https://github.com/polkadot-js/extension/issues/122

### CODE REVIEW

**Comment by jacogr** [comment]
Thank you very much, this is really appreciated. I will take a deep-dive into playing and reading through a bit later today.

**Comment by jacogr** [comment]
My appologies, running behind - really want to get to this one ASAP, just haven't caught a breather. Don't want to keep you handling.

**Comment by jacogr** [comment]
Couple of comments on functionality -

- when I type in the wrong password, we should highlight the box - currently nothing happens, with no indication (just a highlight `isError` is ok as the first generation)

- Instead of copying the contents, rather download it - less error-prone

- we allow the export . of external accounts (i.e. as added via QR), this should not be allowed, we don't have the keys nor a password for these (the same will happen when the extension has ledger access, so on . `isExternal`, don't try and export)

Future -

With the 3rd button on the little bar, we probably want to rethink - move "Forget" and "Export" to the left, keep "Edit" on the right since it is probably the option most used (simple flex with space-between . should do the trick for the 2 groups)

File: __mocks__/extensionizer.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (jacogr) [inline __mocks__/extensionizer.ts]: Brilliant! We are . seriously lacking in tests overall (i.e. we have . none), this is great to make a start.

Just add a copyright header to this file as well.

File: packages/extension-ui/package.json [hunk 2]
```diff
@@ -10,7 +10,13 @@
     "@polkadot/react-qr": "^0.45.0-beta.14",
     "@types/react-router": "^5.1.0",
     "@types/react-router-dom": "^5.1.0",
+    "@types/sinon-chrome": "^2.2.6",
```
Comment (jacogr) [inline packages/extension-ui/package.json]: This is probably also a dev dependency?

File: packages/extension-ui/src/Popup/Export.test.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import React from 'react';
```
Comment (jacogr) [inline packages/extension-ui/src/Popup/Export.test.tsx]: Just add a copyright header, just aligning with the rest.

File: packages/extension-ui/src/Popup/Export.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  };
+
+  return (
+    <div>
+      <Header label='export account' />
+      <Back />
+      <Address address={address}>
+        <Tip header='danger' type='error'>You are exporting your account. Keep it safe and don&apos;t share it with anyone.</Tip>
+        {!passwordEntered && <Input
+          isError={pass.length < MIN_LENGTH}
```
Comment (jacogr) [inline packages/extension-ui/src/Popup/Export.tsx]: As suggested in the other comment, also indicate when the export fails, i.e. we cannot decode.

File: packages/extension-ui/src/Popup/Export.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        />}
+        {!passwordEntered && <Button
+          isDisabled={pass.length === 0}
+          isDanger
+          label='I want to export this account'
+          onClick={_onExportButtonClick}
+          className='export-button'
+          data-export-button
+        />}
+        {passwordEntered && <Tip type='info'>Click below to copy exported JSON to clipboard</Tip>}
```
Comment (jacogr) [inline packages/extension-ui/src/Popup/Export.tsx]: Would move this to download the file.

File: packages/extension-ui/src/messaging.test.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import Adapter from 'enzyme-adapter-react-16';
```
Comment (jacogr) [inline packages/extension-ui/src/messaging.test.ts]: (nit) copyright header above

File: packages/extension/src/background/handlers/Extension.test.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import Extension from '@polkadot/extension/background/handlers/Extension';
```
Comment (jacogr) [inline packages/extension/src/background/handlers/Extension.test.ts]: (nit) Copyright header

File: packages/extension/src/background/handlers/Extension.test.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import keyring from '@polkadot/ui-keyring';
+import { cryptoWaitReady } from '@polkadot/util-crypto';
+import State from '@polkadot/extension/background/handlers/State';
+import { ResponseAccountExport } from '@polkadot/extension/background/types';
+import ExtensionStore from '@polkadot/ui-keyring/stores/Extension';
```
Comment (jacogr) [inline packages/extension/src/background/handlers/Extension.test.ts]: (nit) Just ensure these are in alphabetical order

File: packages/extension-ui/src/Popup/index.tsx [hunk 9]
```diff
@@ -16,6 +16,7 @@ import Accounts from './Accounts';
 import Authorize from './Authorize';
 import Create from './Create';
 import Forget from './Forget';
+import Export from './Export';
```
Comment (jacogr) [inline packages/extension-ui/src/Popup/index.tsx]: (nit) Move one up, alphabetical

**Comment by jacogr** [comment]
Brilliant, thanks for the updates. Will have a final play-through tomorrow.

**Comment by Anze1m** [comment]
Thanks for your appreciation, thanks for your feedback 😁 
I think that I've addressed most of the comments. I've corrected imports, added copyright headers, changed displaying and copying JSON to just downloading it. Furthermore, I've improved password input field behavior and restricted export functionality to be available for non-external accounts only.

**Pull request merged**